### PR TITLE
Initial improvements for bounce handling reporting

### LINKED ISSUE #60: Expose bounce count ratio in the bounce page

### PR DESCRIPTION
**Pull request by bwalkerl**
Closes #60

### CODE REVIEW

File: classes/reportbuilder/local/entities/email_bounce.php [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+    /**
+     * Database tables that this entity uses
+     *
+     * @return string[]
+     */
+    protected function get_default_tables(): array {
+        return [
+            'user_preferences',
+            'user_preferences_send',
```
Comment (bwalkerl) [inline classes/reportbuilder/local/entities/email_bounce.php:46]: Does this need to be a table that exists? I couldn't find an easy way to join with own table and have a separate alias without a second table listed.

File: classes/reportbuilder/local/systemreports/email_bounces.php [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+     * Initialise report, we need to set the main table, load our entities and set columns/filters
+     */
+    protected function initialise(): void {
+        // Our main entity, it contains all of the column definitions that we need.
+        $entitymain = new email_bounce();
+        $entitymainalias = $entitymain->get_table_alias('user_preferences');
+
+        $this->set_main_table('user_preferences', $entitymainalias);
+        $this->add_entity($entitymain);
+        $this->add_base_condition_simple("{$entitymainalias}.name", 'email_bounce_count');
```
Comment (bwalkerl) [inline classes/reportbuilder/local/systemreports/email_bounces.php:45]: Ideally I wanted to add this condition to the entity but couldn't find a way.

**Review by bwalkerl — COMMENTED** [review]
There's a couple of report builder things I was unsure about.

File: classes/reportbuilder/local/entities/email_bounce.php (hunk 1 above)
Comment (brendanheywood) [reply classes/reportbuilder/local/entities/email_bounce.php:46]: I'm not sure, this is the kind of thing I'd ask in the dev chat

**Pull request merged**